dns/namesilo.go

```diff
@@ -145,25 +145,30 @@ func (ns *NameSilo) modify(domain *config.Domain, recordID, recordType, ipAddr s
 	}
 }
 
-func (ns *NameSilo) listRecords(domain *config.Domain) (resp NameSiloDNSListRecordResp, err error) {
-	//lint:ignore SA4006 false positive
+func (ns *NameSilo) listRecords(domain *config.Domain) (*NameSiloDNSListRecordResp, error) {
 	result, err := ns.request("", domain, "", "", nameSiloListRecordEndpoint)
-	err = xml.Unmarshal([]byte(result), &resp)
-	return
+	if err != nil {
+		return nil, err
+	}
+
+	var resp NameSiloDNSListRecordResp
+	if err = xml.Unmarshal([]byte(result), &resp); err != nil {
+		return nil, err
+	}
+
+	return &resp, nil
 }
 
 // request 统一请求接口
 func (ns *NameSilo) request(ipAddr string, domain *config.Domain, recordID, recordType, url string) (result string, err error) {
-	if domain.SubDomain == "@" {
```

Comment:
Ignore those that fill in `@.example.com`?

Reply:
> Ignore those that fill in `@.example.com`?

Already checked here:

https://github.com/jeessy2/ddns-go/blob/d847fb9b4c82e411254def98cffe4afe06b0841a/dns/namesilo.go#L92-L95

So this is unnecessary and can be ignored.